website/content/docs/developers/packing/ffdpacking.md

```diff
@@ -0,0 +1,61 @@
+
+```yaml
+# packing algorithm for packing instances into containers
+heron.class.packing.algorithm:    com.twitter.heron.packing.binpacking.FirstFitDecreasingPacking
+```
+
+The algorithm accepts as input the values of the following parameters:
+
+1. Component RAM 
+2. Hint for the maximum container RAM (`com.twitter.heron.api.Config.TOPOLOGY_CONTAINER_MAX_RAM_HINT`)
+3. Hint for the maximum Container CPU (`com.twitter.heron.api.Config.TOPOLOGY_CONTAINER_MAX_CPU_HINT`)
+4. Hint for the maximum Container Disk (`com.twitter.heron.api.Config.TOPOLOGY_CONTAINER_MAX_DISK_HINT`)
+5. Padding percentage (`com.twitter.heron.api.Config.TOPOLOGY_CONTAINER_PADDING_PERCENTAGE`)
+6. Component Parallelisms
+
+Parameter 1 determines the RAM requirement of each component in the topology.
+If the requirement is not specified then a default value of 1GB is used. The First Fit Decreasing algorithm
+guarantees that the amount of memory allocated to a component is either the one specified by the user
+or the default one.
+
+The parameters 2-4 determine the maximum container size with respect to RAM, cpu cores and disk. 
+If one of these parameters is not specified by the user then
+the hint for the corresponding maximum container resource is set to the default resource requirement of 4 Heron instances. 
+
+Note that these values take into account only the resources allocated for the user's instances. Additional per container resources for system-related processes such as the stream manager can be added to the maximum container size defined above. Thus, the algorithm might eventually produce containers whose size is slighltly bigger that the boundary determined by parameters 2-4.
+The amount of the additional resources allocated to each container to account for additional internal Heron resource requirements, is determined by the padding percentage
+specified in parameter 5. The default value for this parameter is 10, which means that after a container has been filled with user instances, an additional 10% of resources will be allocated to it.
+
+Based on these parameters, the algorithm decides how to place the instances in the containers
+and also how many containers to use. The final result can contain heterogeneous containers. Note that this algorithm does not require the number of containers 
+as input. 
+
+### Configuring the First Fit Decreasing Packing Algorithm
+
+1. The methods `com.twitter.heron.api.Config.setContainerMaxRamHint(long bytes)`, `com.twitter.heron.api.Config.setContainerMaxCpuHint(float ncpus)`, `com.twitter.heron.api.Config.setContainerMaxDiskHint(long bytes)` 
```

Comment:
We should also link to the FirstFitDecreasingPacking javadocs for a detailed explanation of the settings and methods.